### DISCUSSION TITLE
Merging my additions into your fork

### DISCUSSION THREAD

**Discussion by DaniFoldi** [Flowsqy/ShopChest#4] [General]
Hey,

I would like to start with saying thank you for keeping this plugin up-to-date. I have relied heavily on your work to make my fork compatible with 1.18 already.

In the last year or so, I have added a few new features to the plugin, which I would love to get merged into your fork so we don't have two "competing" ones.

The features include:

- More shop materials (all 16+1 shulkers, barrel)
- Per-material tax on selling and buying items
- Push all vault transactions to a separate thread
- Add more item similarity checks (ignore enchantments' order in enchanted items, beehives only care about the number of bees inside, etc.)
- Add a `/shop value` command that shows the average price the item you are holding is sold/bought at
- Probably some other small things I forgot

The reason I'm not just going to open a PR is that because of general laziness I removed some external plugin compatibility layers, the nashorn engine, and pre-1.18 support.

Would you be interested in getting the _good_ changes into your version?

Cheers,
Dani

**Comment by Flowsqy** [comment]
Hey,

First, I'm sorry to answer this late (to be honest, I didn't notice that you have posted something here, I apologize).

Well, I have seen your work, and it's pretty amazing ! I agree with you for the "competitive forks" aspect. Bring changes and improvements in one unique place is better for spigot community.
For all the features you mentioned, there is no reason to refuse to merge them, as they are very coherent for the project. So, to answer you question, YES, I'm interested in getting your changes into my version.
But there is some things that we need to agree on. 
- For the Nashorn engine, I fixed the issue with a custom parser, which is a little bit 'overkill' but does not suppress any feature and improve (at least a little) the performances. 
- For the external plugin compatibility layers, I wanted to move it in an other maven sub project because in my opinion, they should not be in the main class and at random places like they currently are, mostly for code maintainability. I didn't do it because I'm currently super busy.
- For the pre-1.18 support, I really don't know what's is the good decision to make. I don't want to drop it as there is yet people that use it. But I know that things have to move on and keeping all this old code will make the plugin very difficult to update. The way I developed my fork at the beginning intended to don't modify anything but I think that this view over the project is not up to date as the original repository is inactive.

To conclude, I don't know how you want to merge your changes (PR or something else), but I will be very happy to include your work to the project. 

Best regards,

Flowsqy
